pkg/tokens/types/token.go

```diff
@@ -0,0 +1,109 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	tokenKeySeparator = "-"
+)
+
+var (
+	ErrTokenChainNotAllowed   = errors.New("token chain not allowed")
+	ErrInvalidAddressLength   = errors.New("invalid address length")
+	ErrNoSymbol               = errors.New("token has no symbol")
+	ErrDecimalsExceedsMaximum = errors.New("decimals exceeds maximum")
+	ErrInvalidLogoURI         = errors.New("invalid logo URI")
+)
+
+// Token represents a token with cross-chain identification.
+type Token struct {
+	CrossChainID string             `json:"crossChainId"`
```

Comment:
I've avoided adding JSON tags to my structs so far. If/How to serialize/deserialize depends on the consumer (they could use protobuf/XML instead of json, might want different field names even if they wanted JSON, or their use case might not even require serialization at all).

I think the good practice here would be explicitly define serialization in status-go for whatever types are exposed through the API